common/djangoapps/third_party_auth/tests/specs/test_testshib.py

```diff
@@ -207,6 +209,21 @@ def test_configure_testshib_provider_with_cache_duration(self):
         self.assertEqual(num_failed, 0)
         self.assertEqual(len(failure_messages), 0)
 
+    def test_login_with_testshib_provider_short_session_length(self):
+        """
+        Test that when we have a TPA provider which as an explicit maximum
+        session length set, waiting for longer than that between requests
+        results in us being logged out.
+        """
+        # Configure the provider with a 10-second timeout
+        self._configure_testshib_provider(max_session_length=10)
+        # Test the login flow, adding the user in the process
+        super(TestShibIntegrationTest, self).test_login()
+        # Wait 15 seconds; longer than the manually-set 10-second timeout
+        time.sleep(15)
```

Comment:
We should never use `time.sleep()` in tests - it slows down the test suite. Rather use `freezegun` to fake time travel.

![cryolator_ fallout_4](https://cloud.githubusercontent.com/assets/945577/25106851/a6a542c6-2380-11e7-8f5c-825b3383b14d.png)